File: shop-backend-api-app/src/main/java/ru/geekbrains/service/OrderService.java
package ru.geekbrains.service;

import ru.geekbrains.dto.OrderDto;
import ru.geekbrains.dto.ProductDto;
import ru.geekbrains.persist.model.Product;

import java.util.List;

public interface OrderService {

    List<OrderDto> findByUserId(Long userId);

//    List<ProductDto> findProductsByOrder(Long orderId);

    void createOrder(OrderDto orderDto);
}

File: shop-frontend-app/src/app/services/order.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Order} from "../model/order";
import {AllCartDto} from "../model/all-cart-dto";
import {Product} from "../model/product";

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  constructor(private http: HttpClient) { }

  public findOrdersByUser(userId:number) {
    return this.http.get<Order[]>(`/api/v1/order/${userId}`);
  }

  public findProductsByOrder(order: Order) {
    return this.http.get<Product[]>(`/api/v1/order/products/${order.id}`);
  }

  public createOrder(order: Order) {
    this.http.post(`/api/v1/order/create`, order).subscribe();
  }
}
